operator/operator.php

```diff
@@ -57,6 +57,18 @@ public function get_user_categories()
 		// 1.get the categories by unserializing the user object data 'collapsible_categories'
 		// 2.if no categories found, call get_cookie_categories()
 		// 3.return the categories or an empty array
+
+		$collapsible_categories = array();
+		if (!empty($this->user->data['collapsible_categories']))
+		{
+			$collapsible_categories = (array) unserialize($this->user->data['collapsible_categories']);
```

Comment:
This won't really result in an empty array which is what we want to return if unserialize has a problem. This would return `array(0 => false)`